### ISSUE TITLE
[bug report] 'referrerUrl' doesn't have a default value

### ISSUE THREAD

**Issue by tomdavies** [nystudio107/retour#25]
On Retour 1.0.14 / MySQL  5.7.10 , if a user  _directly_  hits a URL that 404s(ie with no referrer set) the following error is thrown:

```
CDbException

CDbCommand failed to execute the SQL statement: SQLSTATE[HY000]: General error: 1364 Field 'referrerUrl' doesn't have a default value. The SQL statement executed was: INSERT INTO `craft_retour_stats` (`redirectSrcUrl`, `hitCount`, `hitLastTime`, `handledByRetour`, `uid`, `dateUpdated`, `dateCreated`) VALUES (:yp0, :yp1, :yp2, :yp3, :yp4, :yp5, :yp6). Bound with :yp0='/dfg', :yp1=1, :yp2='2016-07-11 10:04:00', :yp3=0, :yp4='dc7bd1b0-9fb8-45ca-b588-a77d69902eaa', :yp5='2016-07-11 10:04:00', :yp6='2016-07-11 10:04:00' (/home/vagrant/code/craft/app/framework/db/CDbCommand.php:358)

#0 /home/vagrant/code/craft/app/framework/db/ar/CActiveRecord.php(1081): CDbCommand->execute()
#1 /home/vagrant/code/craft/app/framework/db/ar/CActiveRecord.php(810): CActiveRecord->insert(NULL)
#2 /home/vagrant/code/plugins/retour/services/RetourService.php(333): CActiveRecord->save()
#3 /home/vagrant/code/plugins/retour/RetourPlugin.php(51): Craft\RetourService->incrementStatistics('/dfg', 0)
#4 /home/vagrant/code/craft/app/framework/base/CComponent.php(567): Craft\RetourPlugin->Craft\{closure}(Object(CExceptionEvent))
#5 /home/vagrant/code/craft/app/framework/base/CApplication.php(874): CComponent->raiseEvent('onexception', Object(CExceptionEvent))
#6 /home/vagrant/code/craft/app/framework/base/CApplication.php(745): CApplication->onException(Object(CExceptionEvent))
#7 [internal function]: CApplication->handleException(Object(Craft\HttpException))
#8 {main}
```

**Comment by khalwat** [comment]
So that's very odd, I'm not sure how that could be happening.  The code in the routine it says it is barfing on does fill in that value.

```    public function incrementStatistics($url, $handled = false)
    {

        $handled = (int) $handled;
        $url = substr($url, 0, 255);

/* -- See if a stats record exists already */

        $result = craft()->db->createCommand()
            ->select('*')
            ->from('retour_stats')
            ->where('redirectSrcUrl =' . craft()->db->quoteValue($url))
            ->queryAll();

        if (empty($result))
        {
            $stats = new Retour_StatsRecord;
            $stats->redirectSrcUrl = $url;
            $stats->referrerUrl = craft()->request->getUrlReferrer();
            $stats->hitCount = 1;
            $stats->hitLastTime = DateTimeHelper::currentUTCDateTime();
            $stats->handledByRetour = $handled;
            $stats->save();
        }
        else
        {

/* -- Update the stats table */

            foreach ($result as $stat)
            {
                $stat['hitCount'] = $stat['hitCount'] + 1;
                $stat['hitLastTime'] = DateTimeHelper::currentTimeForDb();
                $stat['referrerUrl'] = craft()->request->getUrlReferrer();

                $result = craft()->db->createCommand()
                    ->update('retour_stats', array(
                        'hitCount' =>  $stat['hitCount'],
                        'hitLastTime' =>  $stat['hitLastTime'],
                        'handledByRetour' =>  $handled,
                        'referrerUrl' =>  $stat['referrerUrl'],
                        ), 'id=:id', array(':id' => $stat['id']));
            }
        }
    } /* -- incrementStatistics */
```

**Comment by khalwat** [comment]
Ugh, `getUrlReferrer` returns `null` if there's no referrer.  Fixed in 1.0.15:

https://github.com/nystudio107/retour

**Comment by tomdavies** [comment]
cheers for the quick fix @khalwat
